packages/markdown/index.js

```diff
@@ -65,49 +64,19 @@ const plugin = {
         const mdFiles = glob.sync(`${mdsInRoute}/**/*.md`);
 
         for (const file of mdFiles) {
-          let md = fs.readFileSync(file, 'utf-8');
-
-          // replace md images if image plugin is being used
-          if (plugin.settings.plugins['@elderjs/plugin-images'] && plugin.config.useElderJsPluginImages) {
-            const MDImgRegex = /!\[([A-Za-z-_ \d]*)\]\(([^)]*)\)/gm;
-            let match;
-            while ((match = MDImgRegex.exec(md)) !== null) {
-              const [fullMatch, alt, src] = match;
-              md = md.replace(
-                fullMatch,
-                `<div class="md-img">${openPattern}picture alt="${alt}" src="${src}" /${closePattern}</div>`,
-              );
-            }
-          }
-
-          const {
-            contents: html,
-            data: { frontmatter, ...data },
-          } = await plugin.markdownParser.process(md);
-          let slug;
-
-          if (plugin.config.slugFormatter && typeof plugin.config.slugFormatter === 'function') {
-            let relativePath = file.replace(`${mdsInRoute}/`, '');
-            slug = plugin.config.slugFormatter(relativePath, frontmatter);
-          }
-          if (typeof slug !== 'string') {
-            if (frontmatter && frontmatter.slug) {
-              slug = frontmatter.slug;
-            } else {
-              slug = file.replace('.md', '').split('/').pop();
-              if (slug.includes(' ')) {
-                slug = slug.replace(/ /gim, '-');
-              }
-            }
-          }
-          plugin.markdown[route].push({
-            slug,
-            frontmatter,
-            html,
-            data,
+          const markdown = createMarkdownStore({
+            root: mdsInRoute,
+            file,
+            parser: plugin.markdownParser,
+            useImagePlugin: plugin.settings.plugins['@elderjs/plugin-images'] && plugin.config.useElderJsPluginImages,
+            shortcodes: plugin.settings.shortcodes,
+            slug: plugin.config.slugFormatter,
           });
+          await markdown.prepareSlug();
```

Comment:
TBH I don't know if this approach does improve performance.

To benchmark, we can change this line to `await markdown.prepareHtml()` to use the old behavior.

Reply:
I believe this should be quiet a bit faster due to not processing all of the html. Also should help out of memory issues on huge sites.